iati/core/tests/test_rulesets.py

```diff
@@ -287,22 +309,36 @@ def test_is_valid_for_raises_error_when_passed_an_etree(self, rule_basic_init):
         with pytest.raises(AttributeError):
             rule_basic_init.is_valid_for(iati.core.resources.load_as_tree(iati.core.resources.get_test_data_path('valid_atleastone')))
 
-    def test_empty_xpath_base_rule_init_normalised_paths(self, rule_empty_xpath_base):
-        """Check that a Rule with an empty xpath_base can be instantiated correctly with normalised `paths`.
+    def test_multiple_valid_context_matches_is_valid_for(self, valid_multiple_context, valid_nest_case, rule_constructor, valid_dataset):
+        """Check Rule returns expected result when checking multiple contexts."""
+        rule = rule_constructor(valid_multiple_context, valid_nest_case)
+        assert rule.is_valid_for(valid_dataset)
+
+    def test_multiple_valid_context_matches_is_invalid_for(self, valid_multiple_context, invalid_nest_case, rule_constructor, invalid_dataset):
+        """Check Rule returns expected result when checking multiple contexts."""
+        rule = rule_constructor(valid_multiple_context, invalid_nest_case)
+        assert not rule.is_valid_for(invalid_dataset)
+
+    def test_condition_case_is_True(self, valid_condition_rule, valid_dataset):
+        """Check that if a condition is `True`, the rule returns None."""
+        assert valid_condition_rule.is_valid_for(valid_dataset) is None
+
+    def test_condition_case_is_False(self, invalid_condition_rule, invalid_dataset):
+        """Check that if a condition is `False`, the rule validates normally.
 
         Note:
-            Rules without `paths` are checked for correct normalisation in their own TestRuleSubclass.
+            Using an invalid dataset so expecting Rules to evaluate to `False`.
 
         """
-        if 'paths' in dir(rule_empty_xpath_base):
-            for path in rule_empty_xpath_base.paths:
-                assert path.startswith(rule_empty_xpath_base.xpath_base)
+        assert not invalid_condition_rule.is_valid_for(invalid_dataset)
 
-    def test_empty_xpath_base_empty_path_is_valid_for(self, rule_empty_xpath_base_and_paths, valid_dataset):
-        """Check that a Rule with an empty xpath_base and empty paths raises error."""
-        # RuleStartswith raises an IndexError instead.
-        with pytest.raises((lxml.etree.XPathEvalError, IndexError)):
-            rule_empty_xpath_base_and_paths.is_valid_for(valid_dataset)
+    @pytest.mark.parametrize("junk_condition", [''] + iati.core.tests.utilities.find_parameter_by_type(['str'], False))
+    def test_uninstantiating_condition_case(self, rule_constructor, valid_single_context, validating_case, junk_condition):
+        """Check that a non-permitted condition case will not instantiate."""
+        junk_condition_case = deepcopy(validating_case)
+        junk_condition_case['condition'] = junk_condition
+        with pytest.raises(Exception):
```

Comment:
Should look for a more specified exception

Reply:
Woops! Forgot to change this one!

Reply:
done